src/Examine/LuceneEngine/Directories/ExamineDirectory.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Lucene.Net.Index;
+using static Lucene.Net.Index.IndexWriter;
+
+namespace Examine.LuceneEngine.Directories
+{
+    public abstract class ExamineDirectory : Lucene.Net.Store.Directory
+    {
+        private Action<ExamineIndexWriter> _onCommitAction;
+        public virtual void InvokeOnCommit(ExamineIndexWriter writer)
+        {
+            _onCommitAction?.Invoke(writer);
+        }
+
+
+        /// <summary>
+        /// Called on commit
+        /// </summary>
+        /// <param name="action"></param>
+        public void SetOnCommitAction(Action<ExamineIndexWriter> action)
+        {
+            _onCommitAction = action;
+        }
+
+        private Func<IndexWriter, MergePolicy> _mergePolicy;
+
+        public virtual bool IsReadOnly { get; set; }
+        public virtual MergePolicy GetMergePolicy(IndexWriter writer)
```

Comment:
The result of this can be null, do we need to check for that null wherever this is used? Same with the other Get... methods here. Not sure if Lucene will explode if a null is passed to the methods where this is used.